pkg/kubelet/cm/cpumanager/cpu_assignment.go

```diff
@@ -606,9 +683,12 @@ func takeByTopologyNUMAPacked(topo *topology.CPUTopology, availableCPUs cpuset.C
 
 	// 2. Acquire whole cores, if available and the container requires at least
 	//    a core's-worth of CPUs.
-	acc.takeFullCores()
-	if acc.isSatisfied() {
-		return acc.result, nil
+	//    If `CPUSortingStrategySpread` is specified, skip taking the whole core.
+	if cpuSortingStrategy != CPUSortingStrategySpread {
+		acc.takeFullCores()
+		if acc.isSatisfied() {
+			return acc.result, nil
+		}
```

Comment:
it would be nice to somehow move this check inside the accumulator function so we don't have checks in this flow concerning the cpuSortingStrategy. Perhaps `acc.takeFullCores()` can bail out right at the beginning or delegate to a helper in sortAvailableCPUsPacked.

Reply:
Since accumulator already knows the strategy, I feel it's reasonable to move inside the accumulator.  Let me take some time to refactor this part.

Reply:
Do you think easily moving `if cpuSortingStrategy != CPUSortingStrategySpread {` check to `acc.takeFullCores()` reasonable? takeFullCores should only be available to pack strategy, if it's spread, then it does nothing.

Reply:
@ffromani kindly ping on above comment.

Reply:
(sorry for late answer) yes, it could be a good way forward. Let's try.